src/lsm/manifest_log.zig

```diff
@@ -756,24 +807,57 @@ pub fn ManifestLogType(comptime Storage: type) type {
             manifest_log.flush(callback);
         }
 
+        pub fn checkpoint_references(
+            manifest_log: *const ManifestLog,
+        ) vsr.SuperBlockManifestReferences {
+            assert(manifest_log.opened);
+            assert(!manifest_log.reading);
+            assert(!manifest_log.writing);
+            assert(manifest_log.write_callback == null);
+            assert(manifest_log.grid_reservation == null);
+            assert(manifest_log.log_block_checksums.count ==
+                manifest_log.log_block_addresses.count);
+            assert(manifest_log.blocks.count == 0);
+            assert(manifest_log.blocks_closed == 0);
+            assert(manifest_log.entry_count == 0);
+
+            if (manifest_log.log_block_addresses.count == 0) {
+                return std.mem.zeroes(vsr.SuperBlockManifestReferences);
```

Comment:
We are inconsistent about "neutral" element of `SuperBlockManifestReferences)`. Here we return zeros, but in `superblock.manifest_references` we use `null`. 

Given that we need to store _something_ on disk, let's maybe get rid of `null` and add `fn empty() bool` method to `SuperBlockManifestReferences)` for branches&asserts?